extensions/resume.md

```diff
@@ -0,0 +1,162 @@
+
+The `draft/resume` capability is advertised by servers that support resuming connections. This capability is requested by clients that reconnect in this way with the `RESUME` command and have the ability to correctly parse the `RESUMED` message as described in this specification. Once the `draft/resume` capability has been negotiated for a client, servers MAY extend the "ping timeout" length for that client (with the expectation that if the client disconnects, it will try to resume its' connection using the method described here).
+
+### Messages
+
+There are two new messages defined by this specification:
+
+- `RESUME`: Indicates that the client wishes to resume a previous connection.
+- `RESUMED`: Indicates that a client has disconnected+reconnected to the server.
+
+Below, we go through both of these.
+
+#### `RESUME` Command
+
+This command is sent before starting SASL authentication, and indicates that the client wishes to resume a previous connection to the server which may still be active. The command is sent with the following format:
+
+    RESUME oldnick [timestamp]
+
+`oldnick` is the nickname the client had when they last disconnected. `timestamp`, if it is given, is a timestamp which indicates when they received the last message from the server on the old connection. This timestamp is used to indicate to other clients how long the disconnection was for, and uses the same format as the IRCv3 `server-time` extension (i.e. `YYYY-MM-DDThh:mm:ss.sssZ`, or in UTC using extended format as specified by ISO 8601:2004(E) 4.3.2).
+
+This command is sent before SASL authentication has successfully completed. After sending this command, the client continues connection registration as usual.
+
+Once SASL reg completes, if the client's able to successfully resume their connection the server sends them a `RESUMED` message (and their nickname is set to `<oldnick>`). If they cannot resume, the server sends the `ERR_CANNOT_RESUME` numeric (and optional additional error numerics).
+
+If successful, the `RESUMED` message is sent to the connecting user before they receive `RPL_WELCOME (001)`. When this message is sent out, the connecting client's nickname is set to be the old client's nick. From this point forward, the combination of old nickname and new connecting client's username+hostname is used to identify the client from that point forward.
+
+Connection resumption is done on a best-effort basis. If the resumption fails clients SHOULD continue with a regular reconnection.
+
+#### `RESUMED` Message
+
+This message is sent when a client successfully resumes a connection. For clients that have not received numerics indicating that their connection registration is completed, this numeric means that their `RESUME` attempt was successful. For registered clients, this indicates that the given client disconnected and reconnected to the server.
+
+This message has the following format:
+
+    :nick!olduser@oldhost RESUMED nick user host [timestamp]
```

Comment:
This shows `nick!olduser@oldhost` yet further below says that they are the new info, `'user' and 'host' are the username and hostname that the new connecting client has`. Shouldn't this be displayed as `nick!newuser@newhost`?

Reply:
Nah, so the source is the old nickname, old username and old hostname, and the new nick/user/host are included as params on the message so clients receiving this message can update it.

Reply:
Ah yes, my bad. It's referring to the params.

Reply:
Clarified the language there with 6451d1f5ef79303708c7d707a0b8f6693106908a, thanks for the note.